evalml/automl/utils.py

```diff
@@ -85,3 +85,29 @@ def tune_binary_threshold(pipeline, objective, problem_type, X_threshold_tuning,
             y_predict_proba = pipeline.predict_proba(X_threshold_tuning)
             y_predict_proba = y_predict_proba.iloc[:, 1]
             pipeline.threshold = objective.optimize_threshold(y_predict_proba, y_threshold_tuning, X=X_threshold_tuning)
+
+
+def check_all_pipeline_names_unique(pipelines):
+    """Checks whether all the pipeline names are unique.
+
+    Arguments:
+        pipelines (list(PipelineBase)): List of pipelines to check if all names are unique.
+
+    Returns:
+          None
+
+    Raises:
```

Comment:
Good catch. No we're not but I saw you do it in the partial_dependence PR and figured it was necessary here since the function doesn't "do anything" but raise an exception ehehe

I think it would be cool if we did this in all parts of our codebase but not sure if the value we'd get justifies the cost we'd have to put in to audit all our functions at this point?